src/main/java/org/jenkinsci/plugins/googleplayandroidpublisher/ApkUploadTask.java

```diff
@@ -62,43 +62,66 @@ protected Boolean execute() throws IOException, InterruptedException {
                         "- Application ID: %s%n", getCredentialName(), applicationId));
         createEdit(applicationId);
 
-        // Get the list of existing APKs and their info
-        List<Apk> existingApks = editService.apks().list(applicationId, editId).execute().getApks();
-        if (existingApks == null) existingApks = Collections.emptyList();
-        for (Apk apk : existingApks) {
-            existingVersionCodes.add(apk.getVersionCode());
+        // Figure out what we're going to upload
+        // TODO: The overall APK vs BUNDLE logic can probably be nicer
+        final AppFileFormat fileFormat = appFilesToUpload.get(0).getFileFormat();
+
+        // Fetch information about the app files that already exist on Google Play
+        List<String> existingAppFileHashes = new ArrayList<>();
+        if (fileFormat == AppFileFormat.BUNDLE) {
```

Comment:
Are you sure we want to separate this based on the incoming file?

Whether we choose to upload an AAB or an APK file has no effect at all on what AAB or APK files already exist on the remote service.  I think we should still be aware of all existing version codes and existing file hashes, regardless of type, right?  I mean, the file hashes wouldn't really make sense that an APK file hash would match an AAB hash -- but a versionCode could.

Reply:
Yeah, you're right.. no need to separate them. Fixed!